tests/test_inference_types.py

```diff
@@ -138,3 +140,25 @@ def test_normalize_keys():
     assert isinstance(instance.maybe_items, list)
     assert len(instance.maybe_items) == 1
     assert isinstance(instance.maybe_items[0], DummyType)
+
+
+def test_optional_are_set_to_none():
+    _types = {v for v in types.__dict__.values() if inspect.isclass(v) and issubclass(v, types.BaseInferenceType)}
+    for _type in _types:
+        parameters = inspect.signature(_type).parameters
+        for parameter in parameters.values():
+            if "Optional" in str(parameter.annotation) or (
+                "Union" in str(parameter.annotation) and "NoneType" in str(parameter.annotation)
+            ):
+                assert parameter.default is None, f"Parameter {parameter} of {_type} should be set to None"
+            else:
+                assert (
+                    parameter.default is inspect.Parameter.empty
+                ), f"Parameter {parameter} of {_type} should not have a default"
```

Comment:
Let's remove this one. I think we only want to test if optional fields are defaulting to None. Jsonschema allows to define default values so if we do so in the future (example: `top_k: float = 1.0`) we won't have to update this (unrelated) test.

```suggestion
```